src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory_Methods.vb

```diff
@@ -12,28 +12,127 @@ Namespace Microsoft.CodeAnalysis.Operations
+                            ' Conversion will be the same BoundCompoundAssignmentTargetPlaceholder as value.LeftOnTheRightOpt
+                            If binaryOperator.Left Is value.LeftOnTheRightOpt OrElse
+                               (binaryOperator.Left.Kind = BoundKind.Conversion AndAlso
+                                DirectCast(binaryOperator.Left, BoundConversion).Operand Is value.LeftOnTheRightOpt) Then
+                                Dim operatorKind = Helper.DeriveBinaryOperatorKind(binaryOperator.OperatorKind, binaryOperator.Left)
+                                Return (OperationKind.CompoundAssignment,
+                                        operatorKind,
+                                        IsChecked:=binaryOperator.Checked,
+                                        InConversion:=identityConversion,
+                                        outConversion,
+                                        RightNode:=binaryOperator.Right,
+                                        OperatorMethod:=Nothing)
+                            End If
+                        ElseIf rightConversion.Operand.Kind = BoundKind.UserDefinedConversion OrElse
+                               rightConversion.Operand.Kind = BoundKind.UserDefinedBinaryOperator Then
+
+                            ' In error scenarios, we can have a standard conversion on top of a UserDefinedBinaryOperator
+                            Dim rightOperatorBinary = If(TryCast(rightConversion.Operand, BoundUserDefinedBinaryOperator),
+                                                         TryCast(DirectCast(rightConversion.Operand, BoundUserDefinedConversion).Operand, BoundUserDefinedBinaryOperator))
+                            If rightOperatorBinary IsNot Nothing Then
+                                Dim inConversion As Conversion = identityConversion
+                                Dim operatorMethod As IMethodSymbol = Nothing
+                                If IsUserDefinedBinaryOperatorCompound(rightOperatorBinary, value.LeftOnTheRightOpt, inConversion, operatorMethod) Then
+                                    Dim operatorKind = Helper.DeriveBinaryOperatorKind(rightOperatorBinary.OperatorKind, leftOpt:=Nothing)
+                                    Return (OperationKind.CompoundAssignment,
+                                            operatorKind,
+                                            IsChecked:=rightOperatorBinary.Checked,
+                                            inConversion,
+                                            outConversion,
+                                            RightNode:=GetUserDefinedBinaryOperatorChildBoundNode(rightOperatorBinary, 1),
+                                            operatorMethod)
+                                End If
+                            End If
                         End If
                 End Select
```

Comment:
>End Select [](start = 16, length = 10)

Please add a Case Else and throw unreachable. Also, please update BoundAssignmentOperator.Validate and assert the expected shape of the right node for a compound assignment.